rust/operator-binary/src/hbase_controller.rs

```diff
@@ -778,6 +788,82 @@ fn build_rolegroup_service(
     })
 }
 
+/// The rolegroup metrics [`Service`] is a service that exposes metrics and a prometheus scraping label.
+pub fn build_rolegroup_metrics_service(
+    hbase: &v1alpha1::HbaseCluster,
+    hbase_role: &HbaseRole,
+    rolegroup: &RoleGroupRef<v1alpha1::HbaseCluster>,
+    resolved_product_image: &ResolvedProductImage,
+) -> Result<Service, Error> {
+    let ports = hbase
+        .metrics_ports(hbase_role)
+        .into_iter()
+        .map(|(name, value)| ServicePort {
+            name: Some(name),
+            port: i32::from(value),
+            protocol: Some("TCP".to_string()),
```

Comment:
suggestion: Use `to_owned` instead.

```suggestion
            protocol: Some("TCP".to_owned()),
```